src/e2e/puppeteer/helpers/dragAndDropThought.ts

```diff
@@ -98,11 +97,17 @@ const dragAndDropThought = async (
 
   if (mouseUp) {
     await page.mouse.up()
-    await waitUntil(() => !document.querySelector('[data-drag-in-progress="true"]'))
 
-    // TODO: Why does drop/DragAndDropThought test fails intermittently without a small delay?
-    // Bullet highlight is visible.
-    await sleep(500)
+    // Wait for drag operation to fully complete by checking:
```

Comment:
Was able to successfully remove the arbitrary delay with proper waiting conditions